tests/Browser/Pages/Backend/Posts/PostDetailTest.php

```diff
@@ -0,0 +1,433 @@
+        });
+    }
+
+
+    /**
+     * Make data has comment
+     *
+     * @return void
+     */
+    public function makeData($row)
+    {
+        $faker = Faker::create();
+
+        $category = factory(Category::class)->create();
+
+        $userIds = DB::table('users')->pluck('id')->toArray();
+
+        $donator = factory(Donator::class)->create([
+            'user_id' => $faker->unique()->randomElement($userIds)
+        ]);
+        $donatorIds = DB::table('donators')->pluck('id')->toArray();
+
+        factory(Book::class)->create([
+            'category_id' => $category->id,
+            'donator_id' => $faker->randomElement($donatorIds),
+        ]);
+        $bookIds = DB::table('books')->pluck('id')->toArray();
+        for ($i = 0; $i <= $row; $i++) {
+         factory(Post::class,1)->create([
+             'user_id' => $faker->randomElement($userIds),
+             'book_id' => $faker->randomElement($bookIds),
+         ]);
+        }
+        $postIds = DB::table('posts')->pluck('id')->toArray();
+        for ($i = 0; $i <= $row; $i++) {
+            factory(Comment::class)->create([
+                'post_id' => $faker->randomElement($postIds),
+                'user_id' => $faker->randomElement($userIds),
+            ]);
+        }
+
+        $comments = DB::table('comments')->get();
+        foreach ($comments as $value) {
+            if (isset($value->post_id)) {
+                factory(Comment::class)->create([
+                    'post_id' => $value->post_id,
+                    'user_id' => $faker->randomElement($userIds),
+                    'parent_id' => $value->id
+                ]);
+            }
+        }
+    }
+
+    /**
+     * Make data has not comment
+     *
+     * @return void
+     */
+    public function makeDataNotComment($row)
+    {
+        $faker = Faker::create();
+
+        $category = factory(Category::class)->create();
+
+        $userIds = DB::table('users')->pluck('id')->toArray();
+
+        $donator = factory(Donator::class)->create([
+            'user_id' => $faker->unique()->randomElement($userIds)
+        ]);
+        $donatorIds = DB::table('donators')->pluck('id')->toArray();
+
+        factory(Book::class)->create([
+            'category_id' => $category->id,
+            'donator_id' => $faker->randomElement($donatorIds),
+        ]);
+
+        $bookIds = DB::table('books')->pluck('id')->toArray();
+
+        for ($i = 0; $i <= $row; $i++) {
+         factory(Post::class,1)->create([
+             'user_id' => $faker->randomElement($userIds),
+             'book_id' => $faker->randomElement($bookIds),
+         ]);
+        }
+    }
+
+    /**
+     * Make data post is status
+     *
+     * @return void
+     */
+    public function makeDataReview($row)
+    {
+        $faker = Faker::create();
+
+        $category = factory(Category::class)->create();
+
+        $userIds = DB::table('users')->pluck('id')->toArray();
```

Comment:
nếu k tạo thêm user nào nữa thì k cần tạo thêm mà chỉ cần dùng $this->user->id là được rồi